*** automake.bak	Thu Jul 26 13:12:18 2001
--- automake	Thu Jul 26 13:17:36 2001
***************
*** 3362,3368 ****
      }
      else
      {
! 	&am_error ("AM_GNU_GETTEXT in \`$configure_ac' but \`ALL_LINGUAS' not defined");
      }
  }
  
--- 3362,3369 ----
      }
      else
      {
! 	&am_error ("AM_GNU_GETTEXT in \`$configure_ac' but po/LINGUAS doesn't exist and \`ALL_LINGUAS' not defined")
! 	    if ! -f "po/LINGUAS";
      }
  }
  
