*** libcroco.orig	2019-02-14 20:54:15.681209898 +0100
--- libcroco	2019-02-14 21:08:00.529990044 +0100
***************
*** 65,71 ****
  no-c++
  
  configure.ac:
! gl_LIBCROCO
  
  Makefile.am:
  AM_CPPFLAGS += $(INCGLIB)
--- 65,81 ----
  no-c++
  
  configure.ac:
! # In libtextstyle, we don't want to use an external libcroco, because its
! # dependencies and their dynamic relocations have an impact on the startup
! # time of a program that is linked with it. As you can see by using
! # 'readelf -r ... | wc -l' and 'readelf -d ... | grep NEEDED':
! # - libcroco.so depends on libxml2.
! #   - libxml2.so has more than 3300 relocations and depends on libicuuc.
! #     - libicuuc.so has more than 4900 relocations and depends on libstdc++.
! #       - libstdc++.so has more than 4600 relocations.
! # These are more than 12800 relocations, to perform at program startup.
! # So, force the use of the included libcroco part.
! gl_LIBCROCO([yes])
  
  Makefile.am:
  AM_CPPFLAGS += $(INCGLIB)
