*** noop-styled-ostream.orig
--- noop-styled-ostream
***************
*** 13,23 ****
  configure.ac:
  
  Makefile.am:
- if !WOE32DLL
  lib_SOURCES += noop-styled-ostream.c
- else
- lib_SOURCES += ../woe32dll/c++noop-styled-ostream.cc
- endif
  # This is a Makefile rule that generates multiple files at once; see the
  # automake documentation, node "Multiple Outputs", for details.
  $(srcdir)/noop-styled-ostream.h : $(top_srcdir)/build-aux/moopp noop-styled-ostream.oo.h noop-styled-ostream.oo.c styled-ostream.oo.h ostream.oo.h
--- 13,19 ----
  configure.ac:
  
  Makefile.am:
  lib_SOURCES += noop-styled-ostream.c
  # This is a Makefile rule that generates multiple files at once; see the
  # automake documentation, node "Multiple Outputs", for details.
  $(srcdir)/noop-styled-ostream.h : $(top_srcdir)/build-aux/moopp noop-styled-ostream.oo.h noop-styled-ostream.oo.c styled-ostream.oo.h ostream.oo.h
