*** gnulib/m4/unlocked-io.m4	2006-04-21 00:28:34.000000000 +0200
--- gettext-tools/m4/unlocked-io.m4	2005-01-20 02:11:33.000000000 +0100
***************
*** 17,24 ****
  
  AC_DEFUN([gl_FUNC_GLIBC_UNLOCKED_IO],
  [
-   AC_LIBSOURCES([unlocked-io.h])
- 
    AC_DEFINE([USE_UNLOCKED_IO], 1,
      [Define to 1 if you want getc etc. to use unlocked I/O if available.
       Unlocked I/O can improve performance in unithreaded apps,
--- 16,21 ----
