*** modules/fnmatch.orig	2011-06-02 22:42:23.000000000 +0200
--- modules/fnmatch	2011-06-02 22:42:43.000000000 +0200
***************
*** 9,15 ****
  
  Depends-on:
  fnmatch-h
! alloca          [test $HAVE_FNMATCH = 0 || test $REPLACE_FNMATCH = 1]
  builtin-expect  [test $HAVE_FNMATCH = 0 || test $REPLACE_FNMATCH = 1]
  flexmember      [test $HAVE_FNMATCH = 0 || test $REPLACE_FNMATCH = 1]
  stdbool         [test $HAVE_FNMATCH = 0 || test $REPLACE_FNMATCH = 1]
--- 9,15 ----
  
  Depends-on:
  fnmatch-h
! alloca-opt      [test $HAVE_FNMATCH = 0 || test $REPLACE_FNMATCH = 1]
  builtin-expect  [test $HAVE_FNMATCH = 0 || test $REPLACE_FNMATCH = 1]
  flexmember      [test $HAVE_FNMATCH = 0 || test $REPLACE_FNMATCH = 1]
  stdbool         [test $HAVE_FNMATCH = 0 || test $REPLACE_FNMATCH = 1]
