*** modules/regex.orig	2011-06-02 22:43:41.000000000 +0200
--- modules/regex	2011-06-02 22:43:53.000000000 +0200
***************
*** 18,24 ****
  btowc           [test $ac_use_included_regex = yes]
  gettext-h       [test $ac_use_included_regex = yes]
  localcharset    [test $ac_use_included_regex = yes]
! malloc-gnu      [test $ac_use_included_regex = yes]
  memcmp          [test $ac_use_included_regex = yes]
  memmove         [test $ac_use_included_regex = yes]
  mbrtowc         [test $ac_use_included_regex = yes]
--- 18,24 ----
  btowc           [test $ac_use_included_regex = yes]
  gettext-h       [test $ac_use_included_regex = yes]
  localcharset    [test $ac_use_included_regex = yes]
! malloc-posix    [test $ac_use_included_regex = yes]
  memcmp          [test $ac_use_included_regex = yes]
  memmove         [test $ac_use_included_regex = yes]
  mbrtowc         [test $ac_use_included_regex = yes]
