*** linebreak.c.orig	2007-05-13 18:54:06.000000000 +0200
--- linebreak.c	2007-05-13 18:54:20.000000000 +0200
***************
*** 455,460 ****
--- 455,462 ----
  }
  
  
+ #ifdef unused
+ 
  /* Determine number of column positions required for first N units
     (or fewer if S ends before this) in S.  */
  
***************
*** 528,533 ****
--- 530,537 ----
    return width;
  }
  
+ #endif
+ 
  
  /* Determine the line break points in S, and store the result at p[0..n-1].  */
  /* We don't support line breaking of complex-context dependent characters
***************
*** 736,741 ****
--- 740,747 ----
      }
  }
  
+ #ifdef unused
+ 
  void
  u16_possible_linebreaks (const unsigned short *s, size_t n, const char *encoding, char *p)
  {
***************
*** 956,961 ****
--- 962,969 ----
      }
  }
  
+ #endif
+ 
  
  /* Choose the best line breaks, assuming the uc_width function.
     Return the column after the end of the string.  */
***************
*** 1044,1049 ****
--- 1052,1059 ----
    return last_column + piece_width;
  }
  
+ #ifdef unused
+ 
  int
  u16_width_linebreaks (const unsigned short *s, size_t n,
                        int width, int start_column, int at_end_columns,
***************
*** 1211,1216 ****
--- 1221,1228 ----
    return last_column + piece_width;
  }
  
+ #endif
+ 
  
  #ifdef TEST1
  
***************
*** 1476,1481 ****
--- 1488,1495 ----
  
  #endif /* C_CTYPE_ASCII */
  
+ #if defined unused || defined TEST2
+ 
  void
  mbs_possible_linebreaks (const char *s, size_t n, const char *encoding,
                           char *p)
***************
*** 1569,1574 ****
--- 1583,1590 ----
      }
  }
  
+ #endif
+ 
  int
  mbs_width_linebreaks (const char *s, size_t n,
                        int width, int start_column, int at_end_columns,
